www/include/options/session/connected_user.php

```diff
@@ -37,18 +37,92 @@
 
-if (isset($_GET["o"]) && $_GET["o"] == "k") {
-    $sid = $_GET['session'];
-    $pearDB->query("DELETE FROM session WHERE session_id = '" . $pearDB->escape($sid) . "'");
+$action = filter_var(
+    $_GET["o"] ?? null,
+    FILTER_SANITIZE_STRING
+);
+
+$selectedUserSid = filter_var(
+    $_GET['session'] ?? null, // the sessionId of the chosen user
+    FILTER_SANITIZE_STRING
+);
+
+$currentPage = filter_var(
+    $_GET['p'] ?? $_POST['p'] ?? 0,
+    FILTER_VALIDATE_INT
+);
+
+if ($selectedUserSid) {
     $msg = new CentreonMsg();
     $msg->setTextStyle("bold");
-    $msg->setText(_("User kicked"));
     $msg->setTimeOut("3");
+
+    switch ($action) {
+        case SYNC_USER:
+            require_once './class/centreonLog.class.php';
+
+            // finding chosen user's data
+            $resUser = $pearDB->prepare(
+                "SELECT `contact_id`, `contact_alias`, `ar_id` FROM contact 
+                LEFT JOIN session ON session.user_id = contact.contact_id 
+                WHERE session.session_id = :userSessionId"
+            );
+            try {
+                $resUser->bindValue(':userSessionId', $selectedUserSid, \PDO::PARAM_STR);
+                $resUser->execute();
+                $currentData = $resUser->fetch();
+
+                // checking if at least one LDAP configuration is still enabled
+                $ldapEnable = $pearDB->query(
+                    "SELECT `value` FROM `options` WHERE `key` = 'ldap_auth_enable'"
+                );
+                $row = $ldapEnable->fetch();
+
+                if ($row['value'] !== '1') {
+                    // unable to plan the manual LDAP request of the contact
+                    $msg->setText(_("No LDAP configuration enabled !"));
+                    break;
+                } else {
+                    // requiring a manual synchronization at next login of the contact
+                    $stmtRequiredSync = $pearDB->prepare(
+                        'UPDATE contact
+                        SET `ldap_required_sync` = "1"
+                        WHERE contact_id = :contactId'
+                    );
+                    $stmtRequiredSync->bindValue(':contactId', $currentData['contact_id'], \PDO::PARAM_INT);
+                    $stmtRequiredSync->execute();
+                    $msg->setTextColor("green");
+                    $msg->setText(_("Resync data requested."));
+                    $msg->setText(_(" ")); // adding the space here, to avoid to forgot it in the translation files
+                    /*
+                     * as every SYNC_USER steps were successful, we need to logout the contact
+                     * to synchronize the data at next login or when the centAcl CRON is executed
+                     */
+                }
+            } catch (\PDOException $e) {
+                $msg->setText(_("Error : unable to read or update the contact data in the DB"));
+                break;
+            }
+        case KICK_USER:
```

Comment:
no break ?!

Reply:
no. It's needed and specified in the comment upper. I'll move the comment on line 119